### PR TITLE
Modules 6 and 7

### PR DESCRIPTION
**Pull request by pldvd**
Pull request introducing:
- JWT authentication (/auth/login endpoint and checkToken middleware)
- test cases

### CODE REVIEW

**Comment by pldvd** [comment]
Screenshot 2023-07-20 at 10 49 27

File: module-2/__tests__/userController.spec.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+  describe('GET /users/:id', () => {
+    it('should return a specific user if exists', async () => {
+      const response = await request(app).get('/users/1');
+
+      expect(response.status).toEqual(StatusCodes.OK);
+      expect(response.body.id).toBe('1');
+      expect(response.body.login).toBe('User1');
+      expect(response.body.age).toEqual(33);
+      expect(response.body.isDeleted).toBe(false);
```
Comment (danielcsato) [inline module-2/__tests__/userController.spec.ts]: https://jestjs.io/docs/expect#tomatchobjectobject

could be simplyfied

**Review by danielcsato — APPROVED** [review]

File: module-2/__tests__/userController.spec.ts (hunk 1 above)
Comment (pldvd) [reply module-2/__tests__/userController.spec.ts]: Thank you for your review! I have simplified it as discussed.

**Pull request merged**